frappe/public/js/frappe/form/controls/base_input.js

```diff
@@ -141,6 +143,17 @@ frappe.ui.form.ControlInput = class ControlInput extends frappe.ui.form.Control
 			(icon ? '<i class="' + icon + '"></i> ' : "") + __(this.df.label) || "&nbsp;";
 		this._label = this.df.label;
 	}
+
+	set_doc_url() {
+		if (!this.df?.doc_url) return;
+
```

Comment:
also might wanna add a condition to ignore buttons, etc where we dont want to show the help thing as even if we ignore them in depends_on, if we change from another fieldtype (say `float`) to `button`, it still shows up 

![Screenshot from 2022-11-17 18-18-54](https://user-images.githubusercontent.com/32034600/202451158-c8d2630a-d61d-4eb9-8667-2a707b4364c4.png)

Reply:
same for grid one as well :P